### PR TITLE
Add weight decay filtering and StableAdamW

### PR DESCRIPTION
**Pull request by warner-benjamin**
This PR adds the [StableAdamW optimizer](https://arxiv.org/abs/2304.13013) as an option. It requires installing [optimi](https://github.com/warner-benjamin/optimi) `pip install torch-optimi`. StableAdamW removes the need for gradient clipping, and I've found it to be a pareto improvement over AdamW.

This PR also adds the `filter_bias_and_bn` option, which prevents weight decay from being applied to linear bias terms and normalization layers. I left it as false to match the current defaults (except in a test) but given its a best practice, we should use it for all of our training.

### CODE REVIEW

**Review by bclavie — APPROVED** [review]
LGTM, thanks!

**Review by orionw — APPROVED** [review]
Also LGTM! Great addition.
